### PR TITLE
Add correct i18n for select2/selectWoo

### LINKED ISSUE #5860: Select2/SelectWoo translations not working
The translations for Select2/SelectWoo are not loaded at all.
https://wordpress.org/support/topic/pods-autocomplete-selectwoo-translate/

Also, the translation strings used in PodsDFV are used correctly.

Incorrect:
`${PodsI18n.__( 'You can only select' )} ${sprintf( PodsI18n._n( '%s item', '%s items', limit ), limit )}`
`${PodsI18n.__( 'Search' )} ${fieldConfig.label}...`

Correct:
`${sprintf( PodsI18n._n( 'You can only select %s item', 'You can only select %s items', limit ), limit )}`
`${sprintf( PodsI18n.__( 'Search %s' ), fieldConfig.label )}...`

### PR DESCRIPTION
**Pull request by JoryHogeveen**
Fixes #5860 

## Description

If there is a translation file available for select2 it will be registered as `pods-select2` and the main file will be registered as `pods-select2-core`. This will make sure the i18n file always gets loaded correctly.

## PR Checklist

- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: classes/PodsI18n.php [hunk 1]
```diff
@@ -176,6 +176,8 @@ private static function default_strings() {
 
 			'Search'                                       => __( 'Search', 'pods' ),
 
+			'Search %s'                                    => __( 'Search %s', 'pods' ),
```
Comment (tr1b0t) [inline classes/PodsI18n.php:179]: :no_entry_sign: A gettext call containing placeholders was found, but was not accompanied by a "translators:" comment on the line above to clarify the meaning of the placeholders.

<sub>hash: bdf703eb98ab59064fb398fec3cf0076</sub>

File: classes/PodsI18n.php [hunk 2]
```diff
@@ -196,6 +198,10 @@ private static function default_strings() {
 
 			'%s items'                                     => __( '%s items', 'pods' ),
 
+			'You can only select %s item'                  => __( 'You can only select %s item', 'pods' ),
```
Comment (tr1b0t) [inline classes/PodsI18n.php:201]: :no_entry_sign: A gettext call containing placeholders was found, but was not accompanied by a "translators:" comment on the line above to clarify the meaning of the placeholders.

<sub>hash: 1243a0622531814d7d96ce6f85541929</sub>

File: classes/PodsI18n.php [hunk 3]
```diff
@@ -196,6 +198,10 @@ private static function default_strings() {
...
+
+			'You can only select %s items'                 => __( 'You can only select %s items', 'pods' ),
```
Comment (tr1b0t) [inline classes/PodsI18n.php:203]: :no_entry_sign: A gettext call containing placeholders was found, but was not accompanied by a "translators:" comment on the line above to clarify the meaning of the placeholders.

<sub>hash: efe013e31dd631fb29594c9cf565b54c</sub>

File: classes/PodsInit.php [hunk 4]
```diff
@@ -524,14 +524,41 @@ public function register_assets() {
...
+			$select2_i18n = PODS_URL . "ui/js/selectWoo/i18n/{$select2_locale}.js";
+		} else {
+			// `en` format.
+			$select2_locale = substr( $select2_locale, 0, 2 );
+			if ( file_exists( PODS_DIR . "ui/js/selectWoo/i18n/{$select2_locale}.js" ) ) {
+				$select2_i18n = PODS_URL . "ui/js/selectWoo/i18n/{$select2_locale}.js";
+			}
+		}
+		if ( $select2_i18n ) {
+			wp_register_script(
```
Comment (tr1b0t) [inline classes/PodsInit.php:542]: :warning: In footer ($in_footer) is not set explicitly wp_register_script; It is recommended to load scripts in the footer. Please set this value to `true` to load it in the footer, or explicitly `false` if it should be loaded in the header.

<sub>hash: 76ef3874421942a057f4eb0cd048763a</sub>

File: classes/PodsInit.php [hunk 5]
```diff
@@ -524,14 +524,41 @@ public function register_assets() {
...
+				'pods-select2-core', PODS_URL . "ui/js/selectWoo/selectWoo{$maybe_min}.js",
```
Comment (tr1b0t) [inline classes/PodsInit.php:543]: :no_entry_sign: Only one argument is allowed per line in a multi-line function call

<sub>hash: bf85d9c30970e707d0589e1f5e38343e</sub>

File: classes/PodsInit.php [hunk 6]
```diff
@@ -524,14 +524,41 @@ public function register_assets() {
...
+				array(
```
Comment (tr1b0t) [inline classes/PodsInit.php:544]: :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: 5376ce2e1feb901602836c2fe24da6d5</sub>

File: classes/PodsInit.php [hunk 7]
```diff
@@ -524,14 +524,41 @@ public function register_assets() {
...
+					'jquery',
+					'pods-i18n',
+				),
+				'1.0.1'
+			);
+			wp_register_script( 'pods-select2', $select2_i18n, array( 'pods-select2-core' ), '1.0.1' );
```
Comment (tr1b0t) [inline classes/PodsInit.php:550]: * :warning: In footer ($in_footer) is not set explicitly wp_register_script; It is recommended to load scripts in the footer. Please set this value to `true` to load it in the footer, or explicitly `false` if it should be loaded in the header.
* :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: abb4fab7736141fd0450d773e4d2c8e8</sub>

File: classes/PodsInit.php [hunk 8]
```diff
@@ -524,14 +524,41 @@ public function register_assets() {
...
+		} else {
+			wp_register_script(
```
Comment (tr1b0t) [inline classes/PodsInit.php:552]: :warning: In footer ($in_footer) is not set explicitly wp_register_script; It is recommended to load scripts in the footer. Please set this value to `true` to load it in the footer, or explicitly `false` if it should be loaded in the header.

<sub>hash: 7d6d1cbab2a11a91ed05284078ea4487</sub>

File: classes/PodsInit.php [hunk 9]
```diff
@@ -524,14 +524,41 @@ public function register_assets() {
...
+				'pods-select2', PODS_URL . "ui/js/selectWoo/selectWoo{$maybe_min}.js",
```
Comment (tr1b0t) [inline classes/PodsInit.php:553]: :no_entry_sign: Only one argument is allowed per line in a multi-line function call

<sub>hash: 1711216333345f11f419bbb816340fa2</sub>

File: classes/PodsInit.php [hunk 10]
```diff
@@ -524,14 +524,41 @@ public function register_assets() {
...
+				array(
```
Comment (tr1b0t) [inline classes/PodsInit.php:554]: :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: 42beebc4079453bc96818f9e908981fa</sub>

**Review by tr1b0t — COMMENTED** [review]
:no_entry_sign: 8 error(s)
:warning: 3 warning(s)

File: classes/PodsInit.php [hunk 11]
```diff
@@ -485,13 +485,17 @@ public function register_assets() {
 		}
 
 		wp_register_script(
-			'pods', PODS_URL . 'ui/js/jquery.pods.js', array(
+			'pods',
+			PODS_URL . 'ui/js/jquery.pods.js',
+			array(
```
Comment (tr1b0t) [inline classes/PodsInit.php:490]: :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: 0f7dcda6994885a99052161a796a4645</sub>

File: classes/PodsInit.php [hunk 12]
```diff
@@ -511,27 +515,61 @@ public function register_assets() {
 
 		if ( ! wp_script_is( 'jquery-ui-timepicker', 'registered' ) ) {
 			wp_register_script(
-				'jquery-ui-timepicker', PODS_URL . 'ui/js/timepicker/jquery-ui-timepicker-addon.min.js', array(
+				'jquery-ui-timepicker',
+				PODS_URL . 'ui/js/timepicker/jquery-ui-timepicker-addon.min.js', array(
```
Comment (tr1b0t) [inline classes/PodsInit.php:519]: * :no_entry_sign: Only one argument is allowed per line in a multi-line function call
* :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: 42f507a98dab971c7de21864497083aa</sub>

File: classes/PodsInit.php [hunk 13]
```diff
@@ -511,27 +515,61 @@ public function register_assets() {
...
+			$select2_locale = substr( $select2_locale, 0, 2 );
+			if ( file_exists( PODS_DIR . "ui/js/selectWoo/i18n/{$select2_locale}.js" ) ) {
+				$select2_i18n = PODS_URL . "ui/js/selectWoo/i18n/{$select2_locale}.js";
+			}
+		}
+		if ( $select2_i18n ) {
+			wp_register_script(
+				'pods-select2-core',
+				PODS_URL . "ui/js/selectWoo/selectWoo{$maybe_min}.js",
+				array(
```
Comment (tr1b0t) [inline classes/PodsInit.php:552]: :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: fb402a0645d1c09d7487bb7f86d90e41</sub>

File: classes/PodsInit.php [hunk 14]
```diff
@@ -511,27 +515,61 @@ public function register_assets() {
...
+					'jquery',
+					'pods-i18n',
+				),
+				'1.0.1',
+				true
+			);
+			wp_register_script( 'pods-select2', $select2_i18n, array( 'pods-select2-core' ), '1.0.1', true );
```
Comment (tr1b0t) [inline classes/PodsInit.php:559]: :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: 48d69e16b335ea1006882a131703a8b4</sub>

File: classes/PodsInit.php [hunk 15]
```diff
@@ -511,27 +515,61 @@ public function register_assets() {
...
+		} else {
+			wp_register_script(
+				'pods-select2',
+				PODS_URL . "ui/js/selectWoo/selectWoo{$maybe_min}.js",
+				array(
```
Comment (tr1b0t) [inline classes/PodsInit.php:564]: :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: 82921402e888dfa33906b6d666749df4</sub>

**Review by tr1b0t — COMMENTED** [review]
:no_entry_sign: 6 error(s)

**Review by sc0ttkclark — APPROVED** [review]

**Pull request merged**